// vitepress@0.16.1/dist/client/app/data.js
import { ref, readonly, computed, inject } from 'vue';
import serializedSiteData from '@siteData';
import { resolveSiteDataByRoute } from '../shared';
import { withBase } from './utils';
export const dataSymbol = Symbol();
export const siteDataRef = ref(parse(serializedSiteData));
function parse(data) {
    return readonly(JSON.parse(data));
}
// hmr
if (import.meta.hot) {
    import.meta.hot.accept('/@siteData', (m) => {
        siteDataRef.value = parse(m.default);
    });
}
// per-app data
export function initData(route) {
    const site = computed(() => resolveSiteDataByRoute(siteDataRef.value, route.path));
    return {
        site,
        theme: computed(() => site.value.themeConfig),
        page: computed(() => route.data),
        frontmatter: computed(() => route.data.frontmatter),
        lang: computed(() => site.value.lang),
        localePath: computed(() => {
            const { locales, lang } = site.value;
            const path = Object.keys(locales).find((lp) => locales[lp].lang === lang);
            return withBase((locales && path) || '/');
        }),
        title: computed(() => {
            return route.data.title
                ? route.data.title + ' | ' + site.value.title
                : site.value.title;
        }),
        description: computed(() => {
            return route.data.description || site.value.description;
        })
    };
}
export function useData() {
    const data = inject(dataSymbol);
    if (!data) {
        throw new Error('vitepress data not properly injected in app');
    }
    return data;
}
